src/components/Divider.tsx

```diff
@@ -1,61 +1,141 @@
+    )
+    return innerThoughtElement ? innerThoughtElement.getBoundingClientRect().width : 0
+  })
+}
 
 /** A custom horizontal rule. */
 const Divider = ({ path, cssRaw }: { path: Path; cssRaw?: SystemStyleObject }) => {
-  const dividerSetWidth = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const dividerRef = useRef<HTMLDivElement>(null)
+  const [dividerWidth, setDividerWidth] = useState<number>(DIVIDER_MIN_WIDTH)
+
+  // Use the custom hook to get necessary data
+  const { dividerId, isOnlyChild, isTableView, children, thoughtsAtSameDepth } = useDividerData(path)
 
   /** Sets the cursor to the divider. */
   const setCursorToDivider = (e: React.MouseEvent | React.TouchEvent) => {
     e.stopPropagation()
     dispatch(setCursor({ path }))
   }
 
-  /** Get the max width of nearby for divider list child elements, add 30 px and set this width for divider. */
-  const setStyle = () => {
-    if (dividerSetWidth.current) {
-      const parentUl = dividerSetWidth.current.closest('ul')
-      const children = parentUl ? (Array.from(parentUl.childNodes) as HTMLElement[]) : []
-      const widths = children.map((child: HTMLElement) => {
-        if (child.querySelector('[data-divider=true]')) return DIVIDER_PLUS_PX
-        const subs = child.getElementsByClassName('subthought') as HTMLCollectionOf<HTMLElement>
-        return subs.length ? subs[0].offsetWidth + DIVIDER_PLUS_PX : DIVIDER_PLUS_PX
-      })
-      const maxWidth = Math.max(...widths)
-      dividerSetWidth.current.style.width = `${maxWidth > DIVIDER_MIN_WIDTH ? maxWidth : DIVIDER_MIN_WIDTH}px`
+  useLayoutEffect(() => {
+    /** Calculates and updates the Divider's width based on sibling thought widths. */
+    const updateDividerWidth = () => {
+      if (!dividerRef.current) return
+
+      // If the divider is the only child and we're not in Table View
+      if (isOnlyChild && !isTableView) {
+        setDividerWidth(DIVIDER_MIN_WIDTH)
+        return
+      }
+
+      let widths: number[] = []
+
+      if (isTableView && isOnlyChild) {
+        // In Table View and divider is the only child
+        if (thoughtsAtSameDepth.length === 0) {
+          setDividerWidth(DIVIDER_MIN_WIDTH)
+          return
+        }
+        widths = getThoughtWidths(thoughtsAtSameDepth)
+      } else {
+        // Non-Table View or Divider is not the only item
+        const siblingThoughts = children.filter(child => child.id !== dividerId && !isDivider(child.value))
+
+        if (siblingThoughts.length === 0) {
+          setDividerWidth(DIVIDER_MIN_WIDTH)
+          return
+        }
+        widths = getThoughtWidths(siblingThoughts)
+      }
+
+      // Determine the maximum width
+      const maxWidth = widths.length > 0 ? Math.max(...widths) : DIVIDER_MIN_WIDTH
+      setDividerWidth(Math.round(Math.max(maxWidth, DIVIDER_MIN_WIDTH)))
     }
-  }
 
-  useEffect(setStyle)
+    updateDividerWidth()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
@ohrytskov - Can you please explain why the dependencies `children`, `dividerId`, and `thoughtsAtSameDepth` are not required?

Reply:
_Possibly_ related to `1. Divider widths do not update`.